descartes_planner/src/sparse_planner.cpp

```diff
@@ -730,6 +758,22 @@ int SparsePlanner::interpolateSparseTrajectory(const SolutionArray& sparse_solut
         if(checkJointChanges(rough_interp,aprox_interp,MAX_JOINT_CHANGE))
         {
           ROS_DEBUG_STREAM("Interpolated point at position "<<pos);
+
+          // look up previous points joint solution
+          const JointTrajectoryPt& last_joint_pt = joint_points_map_.at(cart_points_[pos-1]->getID());
+          std::vector<double> sol;
+          last_joint_pt.getNominalJointPose(std::vector<double>(), *robot_model, sol);
+          // look up dt
+          double dt = timing_cache_[pos].upper_;
+          // check validity of joint motion
+
+          if (!robot_model->isValidMove(sol, aprox_interp, dt))
```

Comment:
Well, it seems to me that ```checkJointChanges``` tests the interpolated joint position against the valid IK for that point. ```isValidMove``` checks to see if that point is achievable from the previous one given the joint velocity constraints of the system.